lib/run/frame.ts

```diff
@@ -12,14 +12,15 @@ let ids = 0;
 
 export interface FrameOptions<T> {
   operation(): Operation<T>;
+  context?: Record<string, unknown>;
   parent?: Frame;
 }
 
 export function createFrame<T>(options: FrameOptions<T>): Frame<T> {
   let { operation, parent } = options;
   let children = new Set<Frame>();
   let block = createBlock(operation);
-  let context = Object.create(parent?.context ?? {});
+  let context = Object.create(options.context ?? parent?.context ?? {});
```

Comment:
If there is a parent, then I still think that the parent context should be the prototype of this context. 

I think what we want is:

```ts
Object.create(parent?.context ?? {}, Object.getOwnPropertyDescriptors(options.context ?? {}))
```